### ISSUE TITLE
Use `$` prefix in variable names for future-proofness

### ISSUE THREAD

**Issue by zspecza** [declandewet/stylus-conventions#8]
First, read this: https://github.com/LearnBoost/stylus/wiki/1.0.0#what-to-change-in-syntax

In that text, there is description of a change to make a  dollar sign (`$`) prefix compulsory for variable names. Perhaps it might be worth changing your variable names to follow this standard for the sake of being future proof.

**Comment by dbox** [comment]
Is there a reason Axis doesn't do this? FWIW I like the $.

@jenius

**Comment by declandewet** [comment]
I'm not even sure if the Stylus dudes ever finalised this - it seems on the new Stylus repo that this link doesn't exist, so that might hint that they changed their minds - as for why Axis doesn't do this, the majority of Axis was written far prior to me ever writing these conventions so it would be a lot to change. @jenius could expand more on why if he wanted to though :)

**Comment by jenius** [comment]
Yeah, I just don't usually use this in my personal style. Don't see a big need for it. If you think it would help the axis codebase to be more clear though I'm not objected to this change!

**Comment by corysimmons** [comment]
@jenius I'm trying to get everyone on the same page with Stylus syntax and `$` was put on the Stylus rc1.0 todo. Might be worth it.

**Comment by dbox** [comment]
I love the cleanliness of not having the `$` but it's just so much more clear with the `$`, especially if you're looking at someone else's code. I think it'd be good for Axis to be ahead of the 1.0 curve, plus, since we're probably going to do a big update to drop nib, might as well do it as part of that update.

That brings us to variable names. @declandewet says hyphenated-variables.. @jenius says camelCase. I don't have a :horse: in that race, personally.

@corysimmons One of those documents was also talking about dropping property lookup for 1.0. Do you know if that's still the plan? Sure hope not ...

**Comment by corysimmons** [comment]
No idea. All I know is Kizu wants to make Stylus more strict and introduce `$` vars https://github.com/stylus/stylus/wiki/1.0.0#what-to-change-in-syntax

I favor dash-case. It's a CSS preprocessor after all.

**Comment by declandewet** [comment]
![](https://infinit.io/_/39WWXgT.png)
![](https://infinit.io/_/Dzb5rQv.png)
![](https://infinit.io/_/N6SgN4R.png)

![](http://36.media.tumblr.com/tumblr_m26nvnNG5o1qkk10ro1_500.jpg)

**Comment by corysimmons** [comment]
I don't even know what property lookup is. Why is it so important?

**Comment by vendethiel** [comment]
```sass
#foo
  height: 50px
  width: @height
```

**Comment by dbox** [comment]
What ^ @vendethiel said, but it becomes much more powerful when used in mixins & functions.

**Comment by dbox** [comment]
Before I start an uproar this IS what he's saying right?

> We should remove the support for property value syntax. Yes. this is feature a lot of people likes, but it makes the code really, really harder to maintain. Really.

property value = property lookup? @kizu

**Comment by corysimmons** [comment]
Yeah I've literally never used that.

**Comment by dbox** [comment]
CORY YOU ARE NOT HELPING

**Comment by corysimmons** [comment]
What kind of cool things do you guys make with it? I can think of...

```stylus
square($size = 10px)
  width: $size
  height: @width
```

**Comment by kizu** [comment]
> property value = property lookup

Nope, `property value` is colonless syntax for things like `width 10px`.

So, nope, we won't drop _property lookup_, do not worry :)

**Comment by dbox** [comment]
oh yes, by all means, then. I think we're all on board with colons :)

**Comment by dbox** [comment]
but no brackets or semicolons EVAR

**Comment by declandewet** [comment]
@corysimmons another more interesting somewhat useful use case (I think the Axis buttons do something similar IIRC):

```stylus
contrast-text()
  b = @background-color || @background
  color: dark(b) ? tint(b, 60%) : shade(b, 60%)
```

@kizu - +1 to not supporting colon-less syntax.

**Comment by jenius** [comment]
I prefer hyphen separation for variables in css, underscore for js. Never camel case. Property lookup is a pretty nice useful feature. We use it in much of the interaction section of axis, honestly it was the thing that initially drew me to stylus. Would be sad if that was dropped for sure. However, it would be great it stylus wasn't dog slow like it is now, so we'll see how much we are paying for it...

**Comment by declandewet** [comment]
colons*

**Comment by dbox** [comment]
I was just basing it by the variable names in Axis most of which are camel case. Sounds like we are all on same page.

**Comment by dbox** [comment]
So, for moving fwd with Axis are we all in agreement that this:

```stylus
hover-darken(percent = 15%, forceUseColor = false)
```

should become:
```stylus
hover-darken($percent = 15%, $force-use-color = false)
```

?

**Comment by jenius** [comment]
:+1:

**Comment by abstracthat** [comment]
:+1:
